### PR TITLE
[BEAM-3436] Fix RetryHttpRequestInitializerTest to use fake clock/sleeper instead of actually waiting.

### PR DESCRIPTION
**Pull request by lukecwik**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/RetryHttpRequestInitializerTest.java [hunk 1]
```diff
@@ -276,7 +277,14 @@ public LowLevelHttpResponse execute() throws IOException {
     // A sample HTTP request to Google Cloud Storage that uses both default Transport and default
```
Comment (akedin) [inline sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/RetryHttpRequestInitializerTest.java]: does the comment need to be updated as well?

Comment (lukecwik) [reply sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/RetryHttpRequestInitializerTest.java]: Updated

**Comment by lukecwik** [comment]
You don't need to use the same clock since nothing is looking at how far the clock advanced but updated regardless.

**Comment by lukecwik** [comment]
https://builds.apache.org/job/beam_PreCommit_Java_GradleBuild/1055/testReport/org.apache.beam.sdk.util/RetryHttpRequestInitializerTest/ shows that the test takes less then a second now.

**Pull request merged**